dependency-resolution/index.md

```diff
@@ -102,3 +102,50 @@ If instead `D11` depends on exactly `1.0.0` and all others depend on `^1.0.0` it
 If instead `D11` depends on exactly `1.0.0` and all other depends on `^1.2.0`, `Dx` will be installed twice: once as `1.6.0` at `node_modules/Dx` and once as `1.0.0` at `node_modules/D11/node_modules/Dx`.
 
 If a lockfile is used by the parent project, the contents of the `node_modules/` directory are, and by extension Node.js' dependency resolution is, entirely reproducible (ignoring installation scripts).
+
+## Python
+### pip
+pip is the package installer for Python and included with modern versions of Python, it is used to find, download, and install packages from [PyPI](https://pypi.org/) and other Python package indexes.
+
+The process of package resolution in pip relies on three key pieces of information: `project name`, `release version` and `dependencies`. The first two pieces of information identify an individual “candidate” for installation, and the third one is used on “on demand” when backtracking algorithm starts to check that particular candidate.
+
+When `pip install` is executed, pip first performs a dependency resolution process by analyzing all the dependencies of the requested packages to determine the most compatible version (according to [version specifiers](https://packaging.python.org/en/latest/glossary/#term-Version-Specifier) and the most recent version). The resolution process might involve backtracking to find the best combination that satisfies all dependency constraints. Only one version of a dependency will get installed.
+
+The resolution process is based on a separate package, [resolvelib](https://pypi.org/project/resolvelib/) which implements an abstract backtracking resolution algorithm. resolvelib implement  [backjumping technique](https://en.wikipedia.org/wiki/Backjumping) in [1.0.0]((https://github.com/sarugaku/resolvelib/blob/main/CHANGELOG.rst)) version which was [vendored](https://pip.pypa.io/en/stable/news/#v23-1) from `pip 23.1`, it can find a set of packages that meet requirements and whose requirements themselves don't conflict.
```

Comment:
```suggestion
The resolution process is based on a separate package, [`resolvelib`](https://pypi.org/project/resolvelib/) which implements an abstract backtracking resolution algorithm. `resolvelib` implements  [backjumping technique](https://en.wikipedia.org/wiki/Backjumping) in [1.0.0]((https://github.com/sarugaku/resolvelib/blob/main/CHANGELOG.rst)) version which was [vendored](https://pip.pypa.io/en/stable/news/#v23-1) from `pip 23.1`, it can find a set of packages that meet requirements and whose requirements themselves don't conflict.
```

Enclosed resolvelib in ` `` ` to emphasize it.